src/github_domain.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2023 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS-IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Domain objects for github entities."""
+
+from __future__ import annotations
+
+import datetime
+
+from typing import Any, Dict, List, Optional, Type
+
+DEFAULT_TIMESTAMP: datetime.datetime = (
+    datetime.datetime(datetime.MINYEAR, 1, 1, tzinfo=datetime.timezone.utc))
+
+
+class Assignee:
+    """A class representing a assignee of a pull request."""
+
+    def __init__(
+        self,
+        name: str,
+        timestamp: datetime.datetime = DEFAULT_TIMESTAMP
+    ) -> None:
```

Comment:
I think a docstring would be helpful here -- explain whether the name is the real name or github username; explain what the timestamp is. (Perhaps renaming would help to clarify.) This is because the natural concept of "assignee" does not include a timestamp, so some explanation is needed.

Reply:
updated. PTAL

Reply:
But which event is the timestamp referring to?

Basically, you are saying that this class (Assignee) has two attributes, username and timestamp. In other words, the username and the timestamp are **properties** of the assignee. But if you think about it in a real-world context, it doesn't make sense for a general event timestamp to be a property of the assignee, right? That's why I don't understand the modelling here, and if you use vague docstrings like "the timestamp when an event happened", it doesn't really add much in terms of meaning.

Reply:
Updated the argument name to `assigned_on_timestamp`. This argument takes the time when the particular assignee was assigned in a PR. So, from the context of an assignee, these two attributes are making a bit of sense to me because one attribute is the username of the assignee, and the other one is the time when the assignee was assigned to a PR. I agree that it was not explained well enough but I tried to explain it this time. Please have a look and let me know if you still think that it needs changes. Thanks!

Reply:
This is much better now. Thanks!